languagetool-core/src/main/java/org/languagetool/rules/spelling/hunspell/HunspellRule.java

```diff
@@ -244,17 +244,14 @@ public RuleMatch[] match(AnalyzedSentence sentence) throws IOException {
           ruleMatch.setType(RuleMatch.Type.UnknownWord);
           String cleanWord2 = cleanWord.substring(dashCorr);
           if (userConfig == null || userConfig.getMaxSpellingSuggestions() == 0 || ruleMatches.size() <= userConfig.getMaxSpellingSuggestions()) {
-            ruleMatch.setLazySuggestedReplacements(() -> {
```

Comment:
The laziness is lost, didn't find a simple way to keep it and still have a match-specific message